src/components/todoview/PendingTodoCard.js

```diff
@@ -1,17 +1,38 @@
+import { getAuth } from "@firebase/auth";
 import Image from "next/image";
 import { useAppContext } from "../../context/AppContext";
 import { addToComplete, addTodoLater, removeTodo } from "../../utils/utils";
 
 function PendingTodoCard({ todoItem }) {
-  const { todoDispatch } = useAppContext();
+  const { todoDispatch, db } = useAppContext();
+  const auth = getAuth();
+  let displayName = "";
+  if (auth.currentUser !== null) {
+    displayName = auth.currentUser.displayName;
+  }
 
-  function handleAction(action) {
+  async function handleAction(action) {
```

Comment:
Try writing individual functions and separate them out so that you can you the same functions in both of the components.